src/control/IntersectionReferenceSystem.ts

```diff
@@ -102,7 +111,14 @@ export class IntersectionReferenceSystem {
    */
   project(length: number): number[] {
     const { curtain } = this.interpolators;
-    const l = (length - this._offset) / this.length;
+    const { calculateDisplacementFromBottom } = this.options;
+
+    let l: number;
```

Comment:
Personally I try to avoid using `let` in my code to prevent potential mutability. Using a ternary expression here instead would allow you to use `const` instead, and also avoid some curlies and semi colons as well :)

Reply:
I don't immediately see a nice way to do that with @HavardNJ's suggested improved version in comment below. Feel free to help me here :D

Reply:
I like the approach. But I don't want to introduce to much functional magic in this codebase as it has not been the preferred way of working in this repo.

Reply:
Storing the result of ((length - offset) / thisLength) in a const tmp then const l = calculateDisplacementFromBottom ? 1-tmp : temp might be a simple solution without using let

Reply:
Might name the variable something more telling then tmp, but you get the idea.

Reply:
did as suggested. Called the tmp normalizedLength.